### PR TITLE
Update Quality Level 1

### PR DESCRIPTION
**Pull request by pablogs9**
TODO:
 - [x] Performance

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/247/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/257/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/248/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/258/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/252/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/262/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/253/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/263/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/256/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/266/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/264/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/274/)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Contribution Guidelines
+
+The following documents constitutes a set of guidelines to which contributors must adhere.
```
Comment (FranFin) [inline CONTRIBUTING.md:3]:
```suggestion
The present document provides a set of guidelines to which contributors must adhere.

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to commit messages to certify that they have the right to submit the code they are contributing to the project according to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+
+## Code Coverage
+
+As stated in [QUALITY.md](QUALITY.md), all contributions to the project must increase line code coverage.
+Because of this, contributors are asked to locally run a coverage assessment that ensures that code coverage has increased when compared to the latest execution of the [nightly coverage CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/coverage/).
+
+## General considerations
+
+Any other contributing policie can be found in [eProsima general guidelines](https://github.com/eProsima/policies/blob/main/CONTRIBUTING.md).
```
Comment (FranFin) [inline CONTRIBUTING.md:36]:
```suggestion
Any other contributing policy can be found in [eProsima general guidelines](https://github.com/eProsima/policies/blob/main/CONTRIBUTING.md).
```

File: PLATFORM_SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Tier 1 platforms are subjected to our unit test suite and other testing tools on a frequent basis including continuous
+integration jobs and nightly jobs.
+Errors or bugs discovered in these platforms are prioritized for correction by the development team.
+Significant errors discovered in Tier 1 platforms can impact release dates and we strive to resolve all known high
+priority errors in Tier 1 platforms prior to new version releases.
+
+## Tier 2
+
+Tier 2 platforms are subject to periodic CI testing which runs both builds and tests with publicly accessible results.
+The CI is expected to be run at least within a week of relevant changes for the current release of **Micro XRCE-DDS Client**.
```
Comment (FranFin) [inline PLATFORM_SUPPORT.md:17]:
```suggestion
The CI is expected to be run at least within a week from the last relevant changes for the current release of the **Micro XRCE-DDS Client**.
```

File: PLATFORM_SUPPORT.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+encouraged.
+Errors may be present in released product versions for Tier 2 platforms.
+Known errors in Tier 2 platforms will be addressed subject to resource availability on a best effort basis and may or
+may not be corrected prior to new version releases.
+One or more entities should be committed to continuing support of the platform.
+
+## Tier 3
+
+Tier 3 platforms are those for which community reports indicate that the release is functional.
+The development team does not run the unit test suite or perform any other tests on platforms in Tier 3.
```
Comment (FranFin) [inline PLATFORM_SUPPORT.md:29]:
```suggestion
The development team does not run the unit test suite nor perform any other tests on platforms in Tier 3.
```

File: PLATFORM_SUPPORT.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+| arm64        | Tier 3 [s]           |                     | Tier 3 [s]         |
+| arm32        | Tier 3 [s]           |                     | Tier 3 [s]         |
+
+" [s] " Compilation from source.
+
+Other Tier 3 OS:
+
+* FreeRTOS
+* Zephyr RTOS
+* Nuttx
```
Comment (FranFin) [inline PLATFORM_SUPPORT.md:47]:
```suggestion
* NuttX
```

File: QUALITY.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
+This document is a declaration of software quality for **eProsima Micro XRCE-DDS Client** based on the guidelines provided in the [ROS 2 REP-2004 document](https://www.ros.org/reps/rep-2004.html).
```
Comment (FranFin) [inline QUALITY.md:1]:
```suggestion
This document is a declaration of software quality for the **eProsima Micro XRCE-DDS Client** based on the guidelines provided in the [ROS 2 REP-2004 document](https://www.ros.org/reps/rep-2004.html).
```

File: QUALITY.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+# Quality Declaration
+
+**eProsima Micro XRCE-DDS Client** is a C99 library that provides a DDS-XRCE implementation for using DDS in eXtreme Resources Constrained Devices. It follows [standard DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/About-DDS-XRCE/).
```
Comment (FranFin) [inline QUALITY.md:5]:
```suggestion
**eProsima Micro XRCE-DDS Client** is a C99 library that provides a DDS-XRCE implementation for using DDS in eXtreme Resources Constrained Devices. It follows the [DDS-XRCE standard](https://www.omg.org/spec/DDS-XRCE/About-DDS-XRCE/).
```

File: QUALITY.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+The public API is documented [in oficial documentation Read the Docs](https://micro-xrce-dds.docs.eprosima.com/en/latest/client.html#api).
+
+### API Stability Policy [1.iv]
+
+**eProsima Micro XRCE-DDS Client** will only break public API between major releases.
+
+### ABI Stability Policy [1.v]
+
+Any ABI break in **eProsima Micro XRCE-DDS Client** will be done between minor versions and it should be clearly stated on the release notes.
```
Comment (FranFin) [inline QUALITY.md:32]:
```suggestion
Any ABI break in **eProsima Micro XRCE-DDS Client** will be done between minor versions and it should be clearly stated in the release notes.
```

File: QUALITY.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+## Change Control Process [2]
+
+The stability of **eProsima Micro XRCE-DDS Client** is ensured through reviews, CI and tests.
+The change control process can be found in [CONTRIBUTING](CONTRIBUTING.md)
+
+All changes to **eProsima Micro XRCE-DDS Client** occur through pull requests that are required to pass all CI tests.
+In case of failure, only maintainers can merge the pull request, and only when there is enough evidence that the failure is unrelated to the change.
+Additionally, all pull requests must have a positive review from one other contributor that did not author the pull request.
```
Comment (FranFin) [inline QUALITY.md:41]:
```suggestion
Additionally, all pull requests must have at least one positive review from another contributor that did not author the pull request.
```

File: QUALITY.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+### Peer Review Policy [2.iii]
+
+All pull requests will be peer-reviewed by at least one other contributor who did not author the pull request. Approval is required before merging.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI to be considered for merging, unless maintainers consider that there is enough evidence that the failure is unrelated to the changes.
+CI testing is automatically triggered by incoming pull requests.
+Current nightly results can be seen here for all supported platforms:
```
Comment (FranFin) [inline QUALITY.md:60]:
```suggestion
Current nightly results for all supported platforms can be checked at the links:
```

File: QUALITY.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+
+### Documentation Policy [2.v]
+
+All pull requests must resolve related documentation changes before merging as stated in [CONTRIBUTING](CONTRIBUTING.md).
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+**eProsima Micro XRCE-DDS Client** has a documented feature list hosted in [Read the Docs](https://micro-xrce-dds.docs.eprosima.com/en/latest/client.html).
```
Comment (FranFin) [inline QUALITY.md:73]:
```suggestion
**eProsima Micro XRCE-DDS Client** has a documented features list hosted by [Read the Docs](https://micro-xrce-dds.docs.eprosima.com/en/latest/client.html).
```

File: QUALITY.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+**eProsima Micro XRCE-DDS Client** includes a complete API Reference which is hosted in [Read the Docs](https://micro-xrce-dds.docs.eprosima.com/en/latest/client.html#api).
+
+### License [3.iii]
+
+The license for **eProsima Micro XRCE-DDS Client** is Apache 2.0, and a summary can be found in each source file.
+A full copy of the license can be found [here](LICENSE).
+
+### Copyright Statements [3.iv]
+
+**eProsima Micro XRCE-DDS Client** copyright holder provide a statement of copyright in each source file.
```
Comment (FranFin) [inline QUALITY.md:86]:
```suggestion
The **eProsima Micro XRCE-DDS Client** copyright holder provides a statement of copyright in each source file.
```

File: QUALITY.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+   This is done to ensure that the **90%** line coverage goal is eventually met.
+2. Line coverage regressions are only permitted if properly justified and accepted by maintainers.
+3. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, mostly reverting the changes and asking for a more thorough testing of the committed changes.
+4. This policy is enforced through the [nightly Micro XRCE-DDS Client CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/).
+
+As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are asked to run a line coverage assessment locally before submitting a PR.
+
+### Performance [4.iv]
+
+**eProsima Micro XRCE-DDS Client** does not provide performance test regarding memory consumption due to the nature of the library.
```
Comment (FranFin) [inline QUALITY.md:119]:
```suggestion
**eProsima Micro XRCE-DDS Client** does provide performance tests regarding memory consumption due to the nature of the library.
```

File: QUALITY.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+### Coverage [4.iii]
+
+[![Coverage](https://img.shields.io/jenkins/coverage/cobertura?jobUrl=http%3A%2F%2Fjenkins.eprosima.com%3A8080%2Fview%2FMicro%2520XRCE%2Fjob%2FMicro-XRCE-DDS-Client%2520Nightly%2520Master%2520Linux)](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/coverage/)
+
+*eProsima Micro XRCE-DDS Client* aims to provide a line coverage **above 90%**.
+*Micro XRCE-DDS Client* code coverage policy comprises:
+1. All contributions to *Micro XRCE-DDS Client* must increase (or at least keep) current line coverage.
+   This is done to ensure that the **90%** line coverage goal is eventually met.
+2. Line coverage regressions are only permitted if properly justified and accepted by maintainers.
+3. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, mostly reverting the changes and asking for a more thorough testing of the committed changes.
```
Comment (FranFin) [inline QUALITY.md:112]:
```suggestion
3. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, most often reverting the changes and asking for a more thorough testing of the committed changes.
```

File: QUALITY.md [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+4. This policy is enforced through the [nightly Micro XRCE-DDS Client CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/).
+
+As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are asked to run a line coverage assessment locally before submitting a PR.
```
Comment (FranFin) [inline QUALITY.md:115]:
```suggestion
As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are required to run a line coverage assessment locally before submitting a PR.
```

File: QUALITY.md [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+Each part of the public API has tests, and new additions or changes to the public API require tests before being added.
+The tests aim to cover typical usage and corner cases.
+
+### Coverage [4.iii]
+
+[![Coverage](https://img.shields.io/jenkins/coverage/cobertura?jobUrl=http%3A%2F%2Fjenkins.eprosima.com%3A8080%2Fview%2FMicro%2520XRCE%2Fjob%2FMicro-XRCE-DDS-Client%2520Nightly%2520Master%2520Linux)](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/coverage/)
+
+*eProsima Micro XRCE-DDS Client* aims to provide a line coverage **above 90%**.
+*Micro XRCE-DDS Client* code coverage policy comprises:
+1. All contributions to *Micro XRCE-DDS Client* must increase (or at least keep) current line coverage.
```
Comment (FranFin) [inline QUALITY.md:109]:
```suggestion
1. All contributions to *Micro XRCE-DDS Client* must increase (or at least keep) the current line coverage.
```

File: QUALITY.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+3. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, mostly reverting the changes and asking for a more thorough testing of the committed changes.
+4. This policy is enforced through the [nightly Micro XRCE-DDS Client CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/).
+
+As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are asked to run a line coverage assessment locally before submitting a PR.
+
+### Performance [4.iv]
+
+**eProsima Micro XRCE-DDS Client** does not provide performance test regarding memory consumption due to the nature of the library.
+These memory consuption tests are under `test/memory` and evaluates the static size (`text`, `bss` and `data` sections) of the library built with different configuration profiles.
+Also, an `stack` consumption analysis is provided by using [Valgrind Massif tool](https://valgrind.org/docs/manual/ms-manual.html).
```
Comment (FranFin) [inline QUALITY.md:121]:
```suggestion
Also, a `stack` consumption analysis is provided by using [Valgrind Massif tool](https://valgrind.org/docs/manual/ms-manual.html).
```

File: QUALITY.md [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+
+All these results are available in [Micro-XRCE-DDS-Client Nightly Master Performance job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Performance/plot/)
+
+Any performance regression detected in *eprosima Micro XRCE-DDS Client* would be analyzed and, in case that it is related to **eProsima Micro XRCE-DDS Client** or could be solved modifying this library, changes could be made to the library in order to revert the performance regression.
```
Comment (FranFin) [inline QUALITY.md:125]:
```suggestion
Any performance regression detected in *eProsima Micro XRCE-DDS Client* would be analysed and, in case it is related to **eProsima Micro XRCE-DDS Client** or could be solved by modifying this library, changes can be made to the library in order to revert the performance regression.
```

File: QUALITY.md [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+
+### Linters and Static Analysis [4.v]
+
+**eProsima Micro XRCE-DDS Client** [code style](https://github.com/eProsima/cpp-style) is enforced using [*uncrustify*](https://github.com/uncrustify/uncrustify).
+Among the CI tests there are tests that ensures that every pull request is compliant with the code style.
```
Comment (FranFin) [inline QUALITY.md:130]:
```suggestion
Among the CI tests, there are tests that ensure that every pull request is compliant with the code style.
```

File: VERSIONING.md [hunk 20]
```diff
@@ -0,0 +1,9 @@
+Versioning policy declaration
+=============================
+
+Starting on v1.0.0, the version numbers for this library will adhere to the versioning policies defined by [Semantic Versioning](https://semver.org/).
+
+This means that API breaks should only happen between major version changes.
+If an ABI break is required, it should be done between minor versions, and it should be clearly stated on the release notes.
```
Comment (FranFin) [inline VERSIONING.md:7]:
```suggestion
If an ABI break is required, it should be done between minor versions, and it should be clearly stated in the release notes.
```

**Review by FranFin — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/273/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/283/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/274/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/284/)

**Review by FranFin — APPROVED** [review]
